telegram/ext/_picklepersistence.py

```diff
@@ -28,11 +28,50 @@
     cast,
 )
 
+from telegram import Bot
 from telegram._utils.types import FilePathInput
+from telegram._utils.warnings import warn
 from telegram.ext import BasePersistence, PersistenceInput
 from telegram.ext._contexttypes import ContextTypes
 from telegram.ext._utils.types import UD, CD, BD, ConversationDict, CDCData
 
+_REPLACED_KNOWN_BOT = "known_bot"
+_REPLACED_UNKNOWN_BOT = "unknown_bot"
+
+
+class BotPickler(pickle.Pickler):
+    def __init__(self, file, bot):
+        super().__init__(file)
+        self._bot = bot
+
+    def persistent_id(self, obj):
+        # print('in pickle persistent_id')
+        if obj is self._bot:
+            # print("found bot to replace", _REPLACED_KNOWN_BOT)
+            return _REPLACED_KNOWN_BOT
+        elif isinstance(obj, Bot):
+            # TODO: test stacklevel
+            warn('Unknown bot instance found. Will be replaced by `None` during pickling',
+                 stacklevel=3)
+            return _REPLACED_UNKNOWN_BOT
+        return None  # pickles as usual
+
+
+class BotUnpickler(pickle.Unpickler):
+    def __init__(self, file, bot):
+        super().__init__(file)
+        self._bot = bot
+
+    def persistent_load(self, pid):
+        # print('in unpickler persistent_id')
+        # print(pid)
+        if pid == _REPLACED_KNOWN_BOT:
```

Comment:
```suggestion
        if isinstance(pid, str) and pid == _REPLACED_KNOWN_BOT:
```
same below - I recall that we had that problem in `insert/replace_bot` when some `__eq__` raises exceptions …

Reply:
> insert/replace_bot when some __eq__ raises exceptions

im not aware of this problem and what how its related here?

and `persistent_load` is only called for bot instances (whenever return something other than `None` in persistent_id`) so this is not an issue?